crates/sui-core/src/authority/authority_store_tables.rs

```diff
@@ -181,14 +187,18 @@ impl AuthorityPerpetualTables {
     }
 
     pub fn get_epoch(&self) -> SuiResult<EpochId> {
-        Ok(self.get_sui_system_state_object()?.epoch)
+        match self.current_epoch.get(&CURRENT_EPOCH_KEY)? {
+            Some(epoch) => Ok(epoch),
+            None => {
+                self.set_epoch(0)?;
```

Comment:
it'd be useful to add a log here in case there's some unexpected data corruption/deletion